src/main/java/functions/Parser.java

```diff
@@ -0,0 +1,139 @@
+                return new DeleteCommand(s);
+            } else if (first.equals("mark")) {
+                int s = Integer.valueOf(words[1]);
+                return new MarkCommand(s);
+            } else if (first.equals("unmark")) {
+                int s = Integer.valueOf(words[1]);
+                return new UnmarkCommand(s);
+            } else if (first.equals("find")) {
+                String search = words[1];
+                return new SearchCommand(search);
+            } else if (first.equals("todo")) {
+                if (input.length() <= 5) {
+                    return new ErrorCommand("The description of a task cannot be empty.\n");
+                } else {
+                    String desc = input.substring(5);
+                    return new AddCommand(desc);
+                }
+            } else if (first.equals("deadline")) {
+                if (input.length() <= 9) {
+                    return new ErrorCommand("The description of a task cannot be empty.\n");
+                } else {
+                    int y = input.indexOf("/by ");
+                    Task t = null;
+                    if (y == -1) {
+                        return new ErrorCommand("Please enter the deadline of the task in the correct format.\n");
+                    } else {
+                        String desc = input.substring(9, y - 1);
+                        String d = input.substring(y + 4);
+                        LocalDateTime date = parseDate(d);
+                        if (date == null) {
+                            return new ErrorCommand("PLease enter the date and time in the correct format. \n");
+                        }
+                        return new AddCommand(desc, date);
+                    }
+                }
+            } else if (first.equals("event")) {
+                if (input.length() <= 6) {
+                    return new ErrorCommand("The description of a task cannot be empty.\n");
+                } else {
+                    int fromIndex = input.indexOf("/from ");
+                    int toIndex = input.indexOf("/to ");
+                    if (fromIndex == -1 || toIndex == -1) {
+                        return new ErrorCommand("Please enter the start and end time" +
+                                " of the task in the correct format.\n");
+                    } else {
+                        String desc = input.substring(6, input.indexOf("/") - 1);
+                        String s = input.substring(fromIndex + 6, toIndex).trim();
+                        String e = input.substring(toIndex + 4).trim();
+                        LocalDateTime start = parseDate(s);
+                        LocalDateTime end = parseDate(e);
+                        if (start == null || end == null) {
+                            return new ErrorCommand("PLease enter the date and time in the correct format. \n");
+                        }
+                        return new AddCommand(desc, start, end);
```

Comment:
Good effort in abstracting out code into specific classes such as AddCommand, which makes the file length shorter and more readable!